### ISSUE TITLE
Smart case by default?

### ISSUE THREAD

**Issue by sharkdp** [chmln/sd#34]
In #7, it was argued that `sd` should support smart case (which is great!) and should enable it by default, because it is also used in `ripgrep` and `fd`.

I personally really like smart-case and did therefore choose it as a default in `fd`. However, I would like to present an argument for why is should (potentially) *not* be the default in `sd`.

Suppose I want to rename a library/module in a given codebase. The name of the module will likely appear in many different variants:

* `awesome_library`: snake_case for things like namespaces, imports, build scripts, etc.
* `AwesomeLibrary`: PascalCase for things like class/struct names, names of types, etc.
* `AWESOME_LIBRARY`: ALL CAPS for things like include guards, constants, etc.
* (`awesomeLibrary`: camelCase for normal variables, unit test names, etc.)

If I am not careful, and use
```
sd -i awesome_library new_name …
```
first, I will also replace all `AWESOME_LIBRARY` strings with the new lower case name. In this particular example, I am probably better of with case-sensitive as a default.

In other words: I think I hardly ever want to replace *both* "old_name" and "OLD_NAME" by "new_name". In most scenarios, I rather want to change "old_name" to "new_name" and "OLD_NAME" to "NEW_NAME".

**Comment by chmln** [comment]
Good points. I've released v0.6 with the new behavior.

**Comment by sharkdp** [comment]
Thank you very much!
